mysql-persistence/src/main/java/com/netflix/conductor/dao/mysql/MySQLExecutionDAO.java

```diff
@@ -0,0 +1,761 @@
+				.addParameter("taskDefName", taskName)
+				.executeScalarList(String.class));
+
+		return getTasks(taskIds);
+	}
+
+	@Override
+	public List<Task> getTasksForWorkflow(String workflowId) {
+		List<String> taskIds = getWithTransaction(c -> c.createQuery(GET_TASKS_FOR_WORKFLOW)
+				.addParameter("workflowId", workflowId)
+				.executeScalarList(String.class));
+
+		return getTasks(taskIds);
+	}
+
+	@Override
+	public String createWorkflow(Workflow workflow) {
+		workflow.setCreateTime(System.currentTimeMillis());
+		return insertOrUpdateWorkflow(workflow, false);
+	}
+
+	@Override
+	public String updateWorkflow(Workflow workflow) {
+		workflow.setUpdateTime(System.currentTimeMillis());
+		return insertOrUpdateWorkflow(workflow, true);
+	}
+
+	@Override
+	public void removeWorkflow(String workflowId) {
+		try {
+			Workflow wf = getWorkflow(workflowId, true);
+
+			//Add to elasticsearch
+			indexer.update(workflowId, new String[]{RAW_JSON_FIELD, ARCHIVED_FIELD}, new Object[]{om.writeValueAsString(wf), true});
+
+			withTransaction(connection -> {
+				removeWorkflowDefToWorkflowMapping(connection, wf);
+				removeWorkflow(connection, workflowId);
+				removePendingWorkflow(connection, wf.getWorkflowType(), workflowId);
+			});
+
+			for(Task task : wf.getTasks()) {
+				removeTask(task.getTaskId());
+			}
+
+		} catch(Exception e) {
+			throw new ApplicationException(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public void removeFromPendingWorkflow(String workflowType, String workflowId) {
+		withTransaction(connection -> removePendingWorkflow(connection, workflowType, workflowId));
+	}
+
+	@Override
+	public Workflow getWorkflow(String workflowId) {
+		return getWorkflow(workflowId, true);
+	}
+
+	@Override
+	public Workflow getWorkflow(String workflowId, boolean includeTasks) {
+		Workflow workflow = getWithTransaction(tx -> readWorkflow(tx, workflowId));
+
+		if (workflow != null) {
+			if (includeTasks) {
+				List<Task> tasks = getTasksForWorkflow(workflowId);
+				tasks.sort(Comparator.comparingLong(Task::getScheduledTime).thenComparingInt(Task::getSeq));
+				workflow.setTasks(tasks);
+			}
+			return workflow;
+		}
+
+		//try from the archive
+		workflow = readWorkflowFromArchive(workflowId);
+
+		if(!includeTasks) {
+			workflow.getTasks().clear();
+		}
+
+		return workflow;
+	}
+
+	@Override
+	public List<String> getRunningWorkflowIds(String workflowName) {
+		Preconditions.checkNotNull(workflowName, "workflowName cannot be null");
+		return getWithTransaction(tx -> tx.createQuery(GET_PENDING_WORKFLOW_IDS).addParameter("workflowType", workflowName).executeScalarList(String.class));
+	}
+
+	@Override
+	public List<Workflow> getPendingWorkflowsByType(String workflowName) {
+		Preconditions.checkNotNull(workflowName, "workflowName cannot be null");
+		return getRunningWorkflowIds(workflowName).stream().map(this::getWorkflow).collect(Collectors.toList());
```

Comment:
I don't have a good sense of how many `Workflow`s this might return, but it seems like adding a DAO method that directly supported returning a `List<Workflow>` would be more efficient than making an individual call for each item in the stream.

Reply:
yes I agree ... but however there is special logic in the `getWorkflow` method that would be a bit heavy to replicate ... the `RedisExecutionDAO` is constructed the same way btw